clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroComputeRequestBuilder.java

```diff
@@ -379,7 +382,24 @@ private void streamingExecuteInternal(
     executed = true;
 
     long preRequestTimeInNS = time.nanoseconds();
-    SchemaAndToString resultSchema = getResultSchema();
+    SchemaAndToString resultSchema;
+    try {
+      resultSchema = getResultSchema();
+    } catch (InvalidVeniceSchemaException e) {
+      // If the schema is invalid, we need to refresh the schema
+      int previousLatestValueSchemaId = this.latestValueSchemaId;
+      latestValueSchemaId = schemaReader.getLatestValueSchemaId();
```

Comment:
Looking into the current implementation of `RouterBackedSchemaReader` I don't think this will actually trigger the refresh in the problematic scenario. We need to either call `getValueSchema` with a newer schema id than current latest schema id or change the implementation. Otherwise `shouldRefreshLatestValueSchemaEntry` in `maybeFetchLatestValueSchemaEntry` will prevent any actual remote fetch from happening.